templates/owncloud/index.ts

```diff
@@ -0,0 +1,99 @@
+import { Output, randomPassword, Services } from "~templates-utils";
+import { Input } from "./meta";
+
+export function generate(input: Input): Output {
+  const services: Services = [];
+  const databasePassword = randomPassword();
+  const redisPassword = randomPassword();
+  const appEnv = [];
+
+  if (input.databaseType === "mariadb") {
+    appEnv.push(
+      `OWNCLOUD_DB_TYPE=mysql`,
+      `OWNCLOUD_DB_NAME=${input.projectName}`,
+      `OWNCLOUD_DB_USERNAME=mariadb`,
+      `OWNCLOUD_DB_PASSWORD=${databasePassword}`,
+      `OWNCLOUD_DB_HOST=${input.projectName}_${input.databaseServiceName}`,
+      `OWNCLOUD_MYSQL_UTF8MB4=true`,
+      `OWNCLOUD_DOMAIN=https://${input.domain}`,
```

Comment:
you can extract any non db env vars outside of the ifs, to avoid writing it twice

Reply:
Tested it and it works completely fine